### DISCUSSION TITLE
Keywords with same prefix in grammar will cause validation error because langium is not greedy. It will not use the longest match. It use the first match instead.

### DISCUSSION THREAD

**Discussion by durianwaffle** [eclipse-langium/langium#1650] [Q&A] [answered]
I have a grammar with 3 keywords with the same prefix as shown below:
1. `A`
2. `AB`
3. `B`

Then it causes false validation error under keyword `AB` because **langium is not greedy**. It will not get the longest keyword as keyword if there are multiple possible rules for the keyword. Instead, it processed keyword `AB` as two separate keywords `A` and `B` but there is not keyword_a then keyword_b grammar in my langium grammar.

How to fix this issue?

**Comment by msujew** [comment, marked as answer]
Hey @durianwaffle,

how did you define your keywords? Can you give a reproducible example? By default, Langium will perform some sorting on the keywords to ensure they appear in the correct order in the lexer definition to ensure that this issue does not appear.
